src/template.rs

```diff
@@ -53,14 +53,15 @@ impl TryFrom<&str> for Template {
         let (has_error, is_dynamic) = StrftimeItems::new(src).fold((false, false), |pair, item| {
             (pair.0 || is_error(&item), pair.1 || is_dynamic(&item))
         });
-        match has_error {
-            true => Err(TemplateError::StrftimeError),
-            false => Ok(Template {
+        if has_error {
+            Err(TemplateError::StrftimeError)
+        } else {
+            Ok(Template {
```

Comment:
It can be shortened via return-early pattern:
```rust
if has_error {
    return Err(TemplateError::StrftimeError);
}
Ok(Template {
```

Reply:
Only when it is in the return position, and even then I'd consider that semantically worse.

Reply:
I know it's largely a matter of taste, yet there's a problem that's solved by early returns (not this case though):

![image](https://user-images.githubusercontent.com/265507/86823167-2dd7d100-c095-11ea-9ae4-edfbcf2c159c.png)

Over time I started favoring early returns instead of `else`s more. I find that `else`s in non-`return` or `break` positions are quite rare, so I use this pattern widely.

Reply:
I used early return in JS all the time, but in Rust we need extra `return` word, so started using it only in rare cases for removing extra padding.